utils/src/main/scala/it/pagopa/interop/commons/utils/AkkaUtils.scala

```diff
@@ -30,29 +29,62 @@ trait AkkaUtils {
     }
   }
 
-  /** Bypasses any bearer content from the HTTP request
-    */
   object PassThroughAuthenticator extends Authenticator[Seq[(String, String)]] {
     override def apply(credentials: Credentials): Option[Seq[(String, String)]] = Some(Seq.empty)
   }
 
-  def getBearer(contexts: Seq[(String, String)]): Try[String]          =
-    contexts.toMap.get(BEARER).toRight(MissingBearer).toTry
-  def getFutureBearer(contexts: Seq[(String, String)]): Future[String] = getBearer(contexts).toFuture
+  // * These are ugly as hell, but they avoid converting each time the whole Seq to a map to extract just a value
+  @inline def fastGetOpt(contexts: Seq[(String, String)])(k: String): Option[String]                                =
+    contexts.find(_._1 == k).map(_._2)
+  @inline private def fastGet(contexts: Seq[(String, String)])(k: String, ex: Throwable): Either[Throwable, String] =
+    fastGetOpt(contexts)(k).toRight(ex)
+
+  // * Ugly as hell, but avoids converting the whole Either to Try before converting to Future
+  @inline private def toFastFutureUUID(either: Either[Throwable, String]): Future[UUID] = either match {
+    case Left(ex) => Future.failed(ex)
+    case Right(s) =>
+      Try(UUID.fromString(s)) match { // * AFAIK this packing/unpacking is unavoidable
+        case Failure(ex2)  => Future.failed(ex2)
+        case Success(uuid) => Future.successful(uuid)
+      }
+  }
 
-  def getUid(contexts: Seq[(String, String)]): Try[String] = contexts.toMap.get(UID).toRight(MissingUserId).toTry
-  def getUidFuture(contexts: Seq[(String, String)]): Future[String] = getUid(contexts).toFuture
+  @inline private def toFuture[T](either: Either[Throwable, T]): Future[T] =
+    either.fold(Future.failed, Future.successful)
```

Comment:
As I wrote, these functions are ugly as hell, but they avoid )or reduce) the implicit class packing/unpacking and the transformation between a lot of layers of different but similarly shaped monads i.e. Try, Either, Future